### PR TITLE
Make the lib work on Android

### PR DESCRIPTION
**Pull request by kebbbnnn**
The lib doesn't work on android unless I disable the ssl cert verification. I'm not really good at this, but if this solution looks stupid, can you suggest any workaround to make this work on android without disabling security features?

### CODE REVIEW

**Comment by SGMcejo** [comment]
How did you get it to work with this change?
If i do that the openssl lib throws an divided by zero error in the `bn_asm.c` file at line 287. Which seems odd to me.

**Comment by kebbbnnn** [comment]
Hi, can you check the version of openssl in the submodule, it works perfectly fine for me with OpenSSL 1.0.2k

File: source/ClientCurl.cpp [hunk 1]
```diff
@@ -288,7 +288,7 @@ void ClientCurl::HandleInfo::configureCurlHandle() {
 
   curl_easy_setopt(handle, CURLOPT_TIMEOUT, 30);
 
-#if __APPLE__
+#if __APPLE__ | ANDROID
```
Comment (drubinstein) [inline source/ClientCurl.cpp]: Change this to `__APPLE__ || ANDROID`

I'd love to change this to 
```
if constexpr __APPLE || ANDROID {}
```
But I assume we have more C++>=11, <17 users than I'd expect.

**Comment by SGMcejo** [comment]
@kebbbnnn 
strange i have the same version

**Comment by cpsauer** [comment]
Hey @kebbbnnn, @drubinstein, isn't it a fairly serious security risk to disable SSL verification? Doesn't that kinda defeat the purpose of SSL--or is there something I'm missing here?

**Pull request merged**